bokehjs/src/coffee/core/selection_manager.ts

```diff
@@ -10,47 +10,152 @@ import {DataSource} from "models/sources/data_source"
 export type Renderer = any
 export type RendererView = any
 
+export type HitTestResult = Selection | null
+
+export abstract class SelectionPolicy extends Model {
+
+  hit_test(geometry: Geometry, renderer_views: RendererView[]): HitTestResult{
+    return null
+  }
+
+  do_selection(hit_test_result: HitTestResult | null, renderer_views: RendererView[], final: boolean, append: boolean): boolean {
+    if (hit_test_result === null) {
+      return false
+    } else {
+      const source = renderer_views[0].model.data_source
+      source.selected.update(hit_test_result, final, append)
+      source.select.emit()
+      return !source.selected.is_empty()
+    }
+  }
+}
+
+SelectionPolicy.prototype.type = "SelectionPolicy"
+
+export class IntersectRenderers extends SelectionPolicy {
+
+  hit_test(geometry: Geometry, renderer_views: RendererView[]): HitTestResult {
+    let hit_test_result_renderers = []
+    for (let r of renderer_views) {
```

Comment:
Both variables (and a few more) can be `const`. The preference should be to use `const`, and use `let` only when a variable is meant to be reassigned (e.g. a loop counter). `const` doesn't relate to the mutability or immutability of the object behind a variable, so you can push to `const arr = []`.

Reply:
Ok, I understand using `const` for arrays and objects now, but is the `let r of renderer_views` not being reassigned?

Reply:
Right, `for (const r of arr)` may seem a bit unintuitive, but technically there is no explicit assignment to `r` (so, `r = ...`) anywhere, so you can safely use `const`. If it was `for (i = 0, ...)` or `while(r = arr.unshift()) ...` then `let` would be a must.